### PR TITLE
Add model redeploying status

### PR DESCRIPTION
**Pull request by emilys314**
Closes https://issues.redhat.com/browse/RHOAIENG-3409

## Description

Makes the status a spinner after editing a model, popover message says "Redeploying".

Basically the status functions will prioritize looking at the target state, since that is more indicative of what is going on. 
I'm not 100% on what the correct message is to display. But the special return case could be removed and then it would default to the state of the target i.e. "Loading". 

image

## How Has This Been Tested?

1. Deploy a model
2. Edit the model and change something (most things will reload it)
3. The status should change to loading (the polling may be slow to update it)

## Test Impact

Jest unit tests could be added to the utils functions, but there aren't any so i didn't add any.

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [X] Included tags to the UX team if it was a UI/UX change.

@vconzola 

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by vconzola** [comment]
@kaedward Would you please take a quick peek at this? I think it looks OK, but you might have a different suggestion for the popover message.

File: frontend/src/pages/modelServing/screens/global/utils.ts [hunk 1]
```diff
@@ -3,24 +3,36 @@ import { getDisplayNameFromK8sResource } from '~/concepts/k8s/utils';
...
+  }
 
-  const lastFailureMessage = is.status?.modelStatus?.lastFailureInfo?.message;
-  const stateMessage = activeModelState ?? targetModelState ?? 'Unknown';
+  if (
+    activeModelState === InferenceServiceModelState.LOADED &&
+    (targetModelState === InferenceServiceModelState.LOADING ||
+      targetModelState === InferenceServiceModelState.PENDING)
+  ) {
+    return 'Redeploying';
```
Comment (christianvogt) [inline frontend/src/pages/modelServing/screens/global/utils.ts:32]: @emilys314 did you get a response yet on suggestion of the new text? If not, we can always do a followup to update the text if everything else is good to go with the PR.

Should add unit tests for this util function.

**Review by christianvogt — COMMENTED** [review]
The jira issue mentions this:
> When activeModelState is in a pending state we will show the spinner without a tooltip.

Seems to me the fallback message right now is always going to be `Unknown` ?

File: frontend/src/pages/modelServing/screens/global/utils.ts (hunk 1 above)
Comment (emilys314) [reply frontend/src/pages/modelServing/screens/global/utils.ts:32]: No updates on the wording, i believe Katie is out until next week

Sure I'll add tests

**Comment by emilys314** [comment]
> The jira issue mentions this:
> 
> > When activeModelState is in a pending state we will show the spinner without a tooltip.
> 
> Seems to me the fallback message right now is always going to be `Unknown` ?

Yeah so the active vs target states don't seem to be well documented anywhere, but here are some tests from kserve 
- https://github.com/search?q=repo%3Akserve%2Fkserve+ModelState&type=code
- https://github.com/kserve/kserve/blob/master/pkg/apis/serving/v1beta1/inference_service_status_test.go#L405

I don't think  `activeModelState` can ever just be pending.

During initial deployment it would have `activeModelState: '', targetModelState: Loading`

The popover text will be hidden if the function returns "", it can do that now because it looks at activeModelState which becomes that commonly. But targetModelState is more up to date so this PR would make it happen less.

However there does seem merit to return an emptry string sometimes so I can make a small update by changing 
`const stateMessage = targetModelState || activeModelState || 'Unknown';`
to
`const stateMessage = (targetModelState || activeModelState) ?? 'Unknown';`

**Review by alexcreasy — APPROVED** [review]
Not tested, but the code looks good - thanks Emily!

**Pull request merged**